tests/unit/type/list.tcl

```diff
@@ -126,30 +83,75 @@ r config set list-max-ziplist-size 1
         assert_equal [r lpop list4] [string repeat c 500]
         assert_equal [r lpop list4] [string repeat b 500]
         assert_equal [r lpop list4] [string repeat a 500]
-    };
+    } {} {needs:debug}
 
     test {plain node check compression with ltrim} {
+        r debug quicklist-packed-threshold 1b
         r lpush list5 [string repeat a 500]
         r linsert list5 after  [string repeat a 500] [string repeat b 500]
         r rpush list5 [string repeat c 500]
         assert_equal [string repeat b 500] [r lindex list5 1]
         r LTRIM list5 1 -1
         r llen list5
-    } {2}
-
+    } {2} {needs:debug}
 
     test {plain node check compression using lset} {
+        r debug quicklist-packed-threshold 1b
         r lpush list6 [string repeat a 500]
         r LSET list6 0 [string repeat b 500]
         assert_equal [string repeat b 500] [r lindex list6 0]
         r lpush list6 [string repeat c 500]
         r LSET list6 0 [string repeat d 500]
         assert_equal [string repeat d 500] [r lindex list6 0]
-    }
+    } {} {needs:debug}
+}
+
+# check functionality of plain nodes using low packed-threshold
+start_server [list overrides [list save ""] ] {
+    # testing lset with combiniations
+    # plain->zip , zip->plain, plain->plain, zip->zip
+    test {plain check LSET with combinations} {
+        r debug quicklist-packed-threshold 5b
+        r RPUSH lst "aa"
+        r RPUSH lst "bb"
+        r lset lst 0 [string repeat d 50001]
+        set s1 [r lpop lst]
+        assert_equal $s1 [string repeat d 50001]
+        r RPUSH lst [string repeat f 50001]
+        r lset lst 0 [string repeat e 50001]
+        set s1 [r lpop lst]
+        assert_equal $s1 [string repeat e 50001]
+        r RPUSH lst [string repeat m 50001]
+        r lset lst 0 "bb"
+        set s1 [r lpop lst]
+        assert_equal $s1 "bb"
+        r RPUSH lst "bb"
+        r lset lst 0 "cc"
+        set s1 [r lpop lst]
+        assert_equal $s1 "cc"
+        r flushdb
+    } {OK} {needs:debug}
```

Comment:
i'm curious. why do you flush at the end of each test? maybe it's better to flush in the beginning of the next one (if needed)

Reply:
before this merge i was using different names for the lists. In order o use same name and since some of the tests are not clearing the list, I have just flush the db. is that a problem ?